playbooks/centurion_teams.yaml

```diff
@@ -0,0 +1,244 @@
+          {{ nfc_pb_disable_log | default(true) }}
+
+
+    - name: update permissions to include newly created teams
+      ansible.builtin.set_fact:
+        team_permissions: |
+          [
+            {% for team in team_permissions %}
+
+                    {
+                      "organization_id": "{{ team.organization_id }}",
+                      "team_name": "{{ team.team_name }}",
+                      "notes": "{{ team.notes }}",
+                      "permissions": 
+                        {{ team.permissions }},
+                      "url": 
+                        {% if team.url is defined %}
+                          "{{ team.url }}",
+                          
+                        {% elif team.url is not defined %}
+                                            
+                          {% for api_values in api_post_teams.results %}
+
+                            {% if api_values.item.organization_id == team.organization_id %}
+
+                              {% if api_values.json.team_name == team.team_name %}
+
+                                "{{ api_values.json.url }}",
+
+                                {% endif %}
+
+                              {% endif %}
+
+                          {% endfor %}
+
+                        {% endif %}
+                    },
+
+            {% endfor %}
+          ]
+      delegate_to: localhost
+      run_once: true
+      no_log: >    # Contains a secret that logging shows
+          {{ nfc_pb_disable_log | default(true) }}
+
+
+    - name: Patch team permissions
+      ansible.builtin.uri:
+        url: |-
+          {{ item.url }}permissions
+        method: PATCH
+        body_format: json
+        body: "{{ item.permissions }}"
+        headers:
+          Authorization: Token {{ lookup('env', 'CENTURION_TOKEN') }}
+        validate_certs: "{{ lookup('env', 'VALIDATE_CENTURION_CERTS') | default(true) | bool }}"
+        status_code:
+          - 200
+      when: >
+             item.url is defined
+      loop: "{{ team_permissions | list }}"
+      delegate_to: localhost
+      run_once: true
+      no_log: >    # Contains a secret that logging shows
+          {{ nfc_pb_disable_log | default(true) }}
+
+
+    - name: Patch team notes
+      ansible.builtin.uri:
+        url: |-
+          {{ item.url }}
+        method: PATCH
+        body_format: json
+        body: |-
+          {
+          "model_notes": "{{ item.notes }}"
+          }
+        headers:
+          Authorization: Token {{ lookup('env', 'CENTURION_TOKEN') }}
+        validate_certs: "{{ lookup('env', 'VALIDATE_CENTURION_CERTS') | default(true) | bool }}"
+        status_code:
+          - 200
+      when: >
+             item.url is defined
+      loop: "{{ team_permissions | list }}"
+      delegate_to: localhost
+      run_once: true
+      no_log: >    # Contains a secret that logging shows
+          {{ nfc_pb_disable_log | default(true) }}
```

Comment:
requires AWX import feature. see https://nofusscomputing.com/projects/ansible/playbooks/awx/#playbook-import-as-awx-ansible-automation-platform-job-template

Reply:
I'll need to check and potentially update this when I have access to AWX again.

Reply:
its the same as [playbooks/itsm_inventory.yaml](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/files/0e24782c3c623c659840eff30ba5c3161ecfa47b#diff-10bead4e664e412780b944f266e65e74db655d17594c79ce8a1110274126aa54) under the `vars` key